libnmstate/nm/checkpoint.py

```diff
@@ -91,10 +43,11 @@ def get_checkpoints():
 
 class CheckPoint(object):
     def __init__(self, timeout=60, autodestroy=True, dbuspath=None):
-        self._manager = nmdbus_manager()
         self._timeout = timeout
-        self._dbuspath = dbuspath
         self._autodestroy = autodestroy
+        self._dbuspath = dbuspath
+        self._mainloop = nmclient.mainloop(refresh=True)
```

Comment:
IMHO it would be nicer to not use a different mainloop just for checkpoint management.